core/local/change.js

```diff
@@ -208,98 +219,98 @@ function isChildUpdate (a /*: LocalChange */, b /*: LocalChange */) { return chi
+  const newChange /*: LocalDirAddition */ = {
     sideName,
     type: 'DirAddition',
     path: e.path,
     stats: e.stats,
     ino: e.stats.ino
   }
-  if (e.old) change.old = e.old
-  if (e.wip) change.wip = e.wip
-  return change
+  if (e.old) newChange.old = e.old
+  if (e.wip) newChange.wip = e.wip
+  return {newChange}
 }
 
-function dirDeletion (e /*: LocalDirUnlinked */) /*: ?LocalDirDeletion */ {
+function dirDeletion (e /*: LocalDirUnlinked */) /*: ?NewChange */ {
   if (!getInode(e)) return
   log.debug({path: e.path}, 'unlinkDir = DirDeletion')
-  const change /*: LocalDirDeletion */ = {
+  const newChange /*: LocalDirDeletion */ = {
     sideName,
     type: 'DirDeletion',
     path: e.path
   }
-  if (e.old) change.old = e.old
-  if (e.old && e.old.ino) change.ino = e.old.ino
-  return change
+  if (e.old) newChange.old = e.old
+  if (e.old && e.old.ino) newChange.ino = e.old.ino
+  return {newChange}
 }
 
-function fileAddition (e /*: LocalFileAdded */) /*: LocalFileAddition */ {
+function fileAddition (e /*: LocalFileAdded */) /*: NewChange */ {
   log.debug({path: e.path}, 'add = FileAddition')
-  const change /*: LocalFileAddition */ = {
+  const newChange /*: LocalFileAddition */ = {
     sideName,
     type: 'FileAddition',
     path: e.path,
     stats: e.stats,
     ino: e.stats.ino,
     md5sum: e.md5sum
   }
-  if (e.old) change.old = e.old
-  if (e.wip) change.wip = e.wip
-  return change
+  if (e.old) newChange.old = e.old
+  if (e.wip) newChange.wip = e.wip
+  return {newChange}
 }
 
-function fileDeletion (e /*: LocalFileUnlinked */) /*: ?LocalFileDeletion */ {
+function fileDeletion (e /*: LocalFileUnlinked */) /*: ?NewChange */ {
   if (!getInode(e)) return
   log.debug({path: e.path}, 'unlink = FileDeletion')
-  const change /*: LocalFileDeletion */ = {
+  const newChange /*: LocalFileDeletion */ = {
     sideName,
     type: 'FileDeletion',
     path: e.path
   }
-  if (e.old) change.old = e.old
-  if (e.old && e.old.ino) change.ino = e.old.ino
-  return change
+  if (e.old) newChange.old = e.old
+  if (e.old && e.old.ino) newChange.ino = e.old.ino
+  return {newChange}
 }
 
-function fileUpdate (e /*: LocalFileUpdated */) /*: LocalFileUpdate */ {
+function fileUpdate (e /*: LocalFileUpdated */) /*: NewChange */ {
   log.debug({path: e.path}, 'change = FileUpdate')
-  const change /*: LocalFileUpdate */ = {
+  const newChange /*: LocalFileUpdate */ = {
     sideName,
     type: 'FileUpdate',
     path: e.path,
     stats: e.stats,
     ino: e.stats.ino,
     md5sum: e.md5sum
   }
-  if (e.old) change.old = e.old
-  if (e.wip) change.wip = e.wip
-  return change
+  if (e.old) newChange.old = e.old
+  if (e.wip) newChange.wip = e.wip
+  return {newChange}
 }
 
-function fileMoveFromUnlinkAdd (sameInodeChange /*: ?LocalChange */, e /*: LocalFileAdded */) /*: * */ {
+function fileMoveFromUnlinkAdd (sameInodeChange /*: ?LocalChange */, e /*: LocalFileAdded */) /*: ?NewChange */ {
```

Comment:
Shouldn't we return a `ChangeReplacement` here since we're replacing an `UnlinkChange` with a `MoveChange`?

Reply:
I thought exactly the same at first, but later discovered it actually wouldn't work.
So the current implementation is currently working *thanks to some weird behavior*.
But I believe it was worth refactoring since this is now easier to spot.

Reply:
Why wouldn't it work? Don't we replace the change in the list anyway?

Reply:
Another option may be to return a `ChangeReplacement` everywhere relevant, but stop removing the replaced change from `LocalChangeMap#changesByPath` with a comment.